app/src/main/java/io/appium/uiautomator2/utils/XMLHierarchy.java

```diff
@@ -131,6 +131,10 @@ private static void visitNode(Node node, HashMap<String, Integer> instances) {
         attrNode.setNodeValue(instance.toString());
         attributes.setNamedItem(attrNode);
 
+        if (androidClass == null || androidClass.length() == 0) {
```

Comment:
This can just be 
```js
if (!androidClass) {
  androidClass = 'android.view.View';
}
```
or even
```js
androidClass = androidClass || 'android.view.View';
```

Reply:
It's a Java syntax or JavaScript ?

Reply:
Oh shoot! Sorry! My mind is stuck in JS. Disregard. :)

Reply:
LOL :)

Reply:
@mstrelex I'd rather use `isBlank` method from apache lib

Reply:
also, it makes sense to add this verification directly into `cleanTagName` helper